### PR TITLE
Implement JDBC ResultSet.getStatement

### PR DESCRIPTION
**Pull request by adamjshook**

### CODE REVIEW

**Comment by adamjshook** [comment]
Wasn't entirely sure how to handle the in-memory result set. It doesn't look like a `Statement` is present for that.  Open to suggestions.

File: presto-jdbc/src/main/java/io/prestosql/jdbc/InMemoryPrestoResultSet.java [hunk 1]
```diff
@@ -26,11 +27,18 @@
 {
     public InMemoryPrestoResultSet(ZoneId sessionTimeZone, List<Column> columns, List<List<Object>> results)
     {
-        super(sessionTimeZone, columns, requireNonNull(results, "results is null").iterator());
+        super(sessionTimeZone, columns, requireNonNull(results, "results is null").iterator(), null);
```
Comment (findepi) [inline presto-jdbc/src/main/java/io/prestosql/jdbc/InMemoryPrestoResultSet.java:30]: let's make it optional

File: presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java [hunk 2]
```diff
@@ -127,8 +127,9 @@
     private final AtomicReference<List<Object>> row = new AtomicReference<>();
     private final AtomicBoolean wasNull = new AtomicBoolean();
     protected final AtomicBoolean closed = new AtomicBoolean();
+    private final Statement statement;
 
-    AbstractPrestoResultSet(ZoneId resultTimeZone, List<Column> columns, Iterator<List<Object>> results)
+    AbstractPrestoResultSet(ZoneId resultTimeZone, List<Column> columns, Iterator<List<Object>> results, Statement statement)
```
Comment (findepi) [inline presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java:132]: also, i'd suggest to make this the first arg/field

**Comment by adamjshook** [comment]
@findepi Updated!  Let me know if you have any more comments.

File: presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java [hunk 3]
```diff
@@ -1101,7 +1104,12 @@ public void moveToCurrentRow()
     public Statement getStatement()
             throws SQLException
     {
-        throw new NotImplementedException("ResultSet", "getStatement");
+        if (statement.isPresent()) {
+            return statement.get();
+        }
+        else {
+            throw new NotImplementedException("ResultSet", "getStatement");
```
Comment (findepi) [inline presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java:1111]: The method _is_ implemented, but the context doesn't allow us to provide a Statement object
(eg some DatabaseMetaData method returning a ResultSet).

```suggestion
            throw new SQLException("Statement not available");
```

File: presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java [hunk 4]
```diff
@@ -1101,7 +1104,12 @@ public void moveToCurrentRow()
     public Statement getStatement()
             throws SQLException
     {
-        throw new NotImplementedException("ResultSet", "getStatement");
+        if (statement.isPresent()) {
+            return statement.get();
+        }
+        else {
```
Comment (findepi) [inline presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java:1110]: `else` is redundant, since the "then" block always returns

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
CI failed -- https://github.com/prestosql/presto/issues/4973

**Comment by findepi** [comment]
Merged, thanks!

**Pull request merged**